pex/requirements.py

```diff
@@ -56,6 +56,11 @@ def from_url(
         is_constraints=False,  # type: bool
     ):
         # type: (...) -> Iterator[Source]
+        pex_warnings.warn(
```

Comment:
N.B.: Because of this feature, I can't remove URLFetcher in this PR - it still may be used by someone out there to `-r https://my.com/requirements.txt`. Since `pip download` will not fetch non-Python package archives, retaining URLFetcher is the only option until a Pex major version bump.